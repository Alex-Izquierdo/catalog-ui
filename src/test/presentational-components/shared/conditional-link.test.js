import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter, Link } from 'react-router-dom';
import ConditionalLink from '../../../presentational-components/shared/conditional-link';

describe('<ConditionalLink', () => {
  it('should render children wrapped in router Link component', () => {
    const wrapper = mount(
      <MemoryRouter>
        <ConditionalLink to="/some/url">
          <div>Link children</div>
        </ConditionalLink>
      </MemoryRouter>
    );
    expect(wrapper.find(Link)).toHaveLength(1);
    expect(toJson(wrapper.find(ConditionalLink))).toMatchSnapshot();
  });

  it('should render only children withouth link', () => {
    const wrapper = mount(
      <MemoryRouter>
        <ConditionalLink>
          <div>Link children</div>
        </ConditionalLink>
      </MemoryRouter>
    );
    expect(wrapper.find(Link)).toHaveLength(0);
    expect(toJson(wrapper.find(ConditionalLink))).toMatchSnapshot();
  });
});
