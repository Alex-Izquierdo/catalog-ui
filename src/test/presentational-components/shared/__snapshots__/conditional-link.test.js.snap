// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConditionalLink /> should render children wrapped in router Link component 1`] = `
<ConditionalLink
  pathname="/some/url"
>
  <CatalogLink
    nav={false}
    pathname="/some/url"
    preserveSearch={false}
    searchParams={Object {}}
  >
    <Styled(Link)
      to={
        Object {
          "pathname": "/some/url",
          "search": "",
        }
      }
    >
      <Link
        className="sc-AxjAm jfRa-DE"
        to={
          Object {
            "pathname": "/some/url",
            "search": "",
          }
        }
      >
        <LinkAnchor
          className="sc-AxjAm jfRa-DE"
          href="/some/url"
          navigate={[Function]}
        >
          <a
            className="sc-AxjAm jfRa-DE"
            href="/some/url"
            onClick={[Function]}
          >
            <div>
              Link children
            </div>
          </a>
        </LinkAnchor>
      </Link>
    </Styled(Link)>
  </CatalogLink>
</ConditionalLink>
`;

exports[`<ConditionalLink /> should render only children withouth link 1`] = `
<ConditionalLink>
  <div>
    Link children
  </div>
</ConditionalLink>
`;
